src/Dapr.Client/CryptographyOptions.cs

```diff
@@ -0,0 +1,52 @@
+﻿#nullable enable
+namespace Dapr.Client
+{
+    /// <summary>
+    /// A collection of options used to configure how encryption cryptographic operations are performed.
+    /// </summary>
+    public class EncryptionOptions
+    {
+        /// <summary>
+        /// Creates a new instance of the <see cref="EncryptionOptions"/>.
+        /// </summary>
+        /// <param name="keyWrapAlgorithm"></param>
+        public EncryptionOptions(KeyWrapAlgorithm keyWrapAlgorithm)
+        {
+            KeyWrapAlgorithm = keyWrapAlgorithm;
+        }
+
+        /// <summary>
+        /// The name of the algorithm used to wrap the encryption key.
+        /// </summary>
+        public KeyWrapAlgorithm KeyWrapAlgorithm { get; set; }
+
+        /// <summary>
+        /// The size of the block in bytes used to send data to the sidecar for cryptography operations.
+        /// </summary>
+        /// <remarks>
+        /// This defaults to 4KB and generally should not exceed 64KB.
+        /// </remarks>
+        public uint StreamingBlockSizeInBytes { get; set; } = 4 * 1024;
```

Comment:
I generally prefer to apply defaults in the implementation rather than in the options object (i.e. `public uint? StreamingBlockSizeInBytes { get; set }`), else it can be difficult to know whether a value was explicitly set or not which is sometimes helpful in the implementation.

Reply:
It also helps enable the API to allow omitting an options object altogether for simple cases.

Reply:
See, I've typically taken the other route. For a simple method like these, only letting the method be configurable through an options object, itself the sole source of default values and the variations of constructors and validations ensuring the right combination of properties are set for any given configuration, it makes the options class the sole source of truth instead of hunting around in the implementations.

Putting the default values in the implementation then to be overridden by anything coming from the options provides the opportunity for unverified conflicts and only serves to lengthen the implementation which should be focused on whatever it's doing instead of refereeing which options to use.

In my approach, the options object is easily omitted by just adding an overload without it and in the override method, create a new instance of it (knowing all the defaults are set there as expected).

If it's a Dapr convention to put the defaults in the implementation instead of the options, I'll of course abide by it, but I'd otherwise rather leave it as-is.